src/core/lib/gpr/tls_stdcpp.h

```diff
@@ -24,14 +24,7 @@
 /** Thread local storage based on C++ thread_local.
    #include tls.h to use this - and see that file for documentation */
 
-/** Use GPR_TLS_DECL to declare tls static variables outside a class */
-#define GPR_TLS_DECL(name) thread_local static intptr_t name = 0
-
-/** Use GPR_TLS_CLASS_DECL to declare tls static variable members of a class.
- *  GPR_TLS_CLASS_DEF needs to be called to define this member. */
-#define GPR_TLS_CLASS_DECL(name) thread_local static intptr_t name
-
-#define GPR_TLS_CLASS_DEF(name) thread_local intptr_t name = 0
+#define GPR_THREAD_LOCAL(type) thread_local type
```

Comment:
could we make this `static_assert(sizeof(type) == sizeof(intptr_t)); thread_local type`?

I'd like the config that I edit code with to have the same restrictions as the most restrictive platform we build for (otherwise my life is waiting for CI to come back for minor edits)

Reply:
Good idea!

Prepending a static assertion doesn't quite work because then `static GPR_THREAD_LOCAL(std::string) greeting` expands to `static static_assert(...); thread_local std::string greeting` which doesn't compile. Moving the static assertion to the end of the expression will produce something that compiles today but will prevent initializers.

I might have to "pull in" the static modifier into the macro to get this working. Any ideas I haven't thought of?